index.js

```diff
@@ -186,6 +192,9 @@ Router.prototype.handle = function handle(req, res, callback) {
   req.baseUrl = parentUrl
   req.originalUrl = req.originalUrl || req.url
 
+  // trigger the "beginning of route handling" event
+  if (events && routerEvents[0] in events) self.emit('handlestart', req)
```

Comment:
Is `routerEvents[0] in events` actually saving any time? I'm not sure if accessing `_events` and using `in` on it is really a good long term API to use with event emitters.